### DISCUSSION TITLE
Use a css code chunk to embed the CSS rules directly in your quarto document

### DISCUSSION THREAD

**Discussion by ManuelSpinola** [quarto-dev/quarto-cli#1402] [Q&A] [answered]
Is it possible to use a css code chunk to embed the CSS rules directly in your quarto document as in an rmarkdown document?

````
```{css, echo=FALSE}
p {
  font-size: 32px;
}
```
````

**Comment by mcanouil** [comment]
With raw html cell:

````
```{=html}
<style>
...
</style>
```
````

<img width="1624" alt="VSCode screenshot showing a quarto document with inline css and the rendered HTML document" src="https://user-images.githubusercontent.com/8896044/179179753-1481fcd3-4fdd-4f1a-b9ac-06572e4cb4d1.png">

**Comment by cderv** [comment, marked as answer]
@ManuelSpinola yes any **knitr** feature will work with Quarto when you are **knitr** computation engine

````markdown
---
format: html
---

```{css, echo = FALSE}
h1 {
  color: red;
}
```

# Red header 

Some content

```{r}
1 + 1
```
````

![image](https://user-images.githubusercontent.com/6791940/179188873-aebae58b-4599-4532-bc74-57bb08517deb.png)

If you try without any R chunk inside the document, you need to explicitly configure the use of **knitr** engine

````
---
format: htlm
engine: knitr
---
````

See about [engine binding](https://quarto.org/docs/computations/execution-options.html#engine-binding)


However, **knitr** will indeed write the same content from the chunk as @mcanouil answer - so directly writing raw block is a solution. 

For styling with Quarto, using the theme file stays the recommended way: https://quarto.org/docs/output-formats/html-themes.html#theme-options
